src/components/Instance/Types/String/index.svelte

```diff
@@ -1,11 +1,48 @@
-<div>TODO</div>
-
 <script>
 
+    console.log($$props);
+  export let definition = {};
+  let minLength, maxLength, pattern;
+
+  ({ minLength, maxLength, pattern } = {...definition });
+
+  function validNumber(n) { 
+    if (isNaN(n) || n < 0 ) {
+        return "invalid";
+    } 
+    return "valid";
+  };
+
 </script>
 
 <style>
-div {
-  background-color: red;
-}
+  .invalid {
+    background-color: pink;
+  }
+  .valid {
+    background-color: white;
+  }
+  .default {
+    background-color: grey;
+  }
 </style>
+
+<div>
+
+  {#if minLength}
+    <div class="{validNumber(minLength)}" >minLength {minLength}</div>
```

Comment:
Svelte has a cute shortcut for classes. Assuming `validNumber` returns `false` when it's not valid, you could do:

```
<div class:invalid={!validNumber(minLength)} >
```

Reply:
Also, create a story where we have an invalid minLength, so that we can see an example of your sanity code in action. :-)